File: python/6 kyu/338.py
# https://www.codewars.com/kata/546dba39fa8da224e8000467/train/python

# Passed

def run_length_encoding(s):
    count = 1
    result = []
    for idx in range(len(s)):
        if idx != len(s) - 1 and s[idx] == s[idx + 1]:
            count += 1
        else:
            result.append([count, s[idx]])
            count = 1
            
    return result

output = run_length_encoding("hello world!")
print(output)

File: nodejs/8 kyu/338.js
// https://www.codewars.com/kata/5720a1cb65a504fdff0003e2/train/javascript

// Passed

function differenceInAges(ages) {
  let sortedAges = ages.sort((a, b) => a - b);
  let youngestAge = sortedAges[0];
  let oldestAge = sortedAges[sortedAges.length - 1];
  let result = [youngestAge, oldestAge, oldestAge - youngestAge];
  return result;
}

const output = differenceInAges([82, 15, 6, 38, 35]);
console.log(output);